src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -305,16 +315,17 @@
 (define-private (add-pox-addr-to-reward-cycles (pox-addr (tuple (version (buff 1)) (hashbytes (buff 20))))
                                                (first-reward-cycle uint)
                                                (num-cycles uint)
-                                               (amount-ustx uint))
-  (let ((cycle-indexes (list u0 u1 u2 u3 u4 u5 u6 u7 u8 u9 u10 u11)))
+                                               (amount-ustx uint)
+                                               (stacker principal))
+  (let ((cycle-indexes (list u0 u1 u2 u3 u4 u5 u6 u7 u8 u9 u10 u11))
+        (results (fold add-pox-addr-to-ith-reward-cycle cycle-indexes 
+                         { pox-addr: pox-addr, first-reward-cycle: first-reward-cycle, num-cycles: num-cycles, 
+                           reward-set-indexes: (list), amount-ustx: amount-ustx, i: u0, stacker: (some stacker) })))
     ;; For safety, add up the number of times (add-principal-to-ith-reward-cycle) returns 1.
     ;; It _should_ be equal to num-cycles.
-    (asserts! 
-     (is-eq num-cycles 
-                 (get i (fold add-pox-addr-to-ith-reward-cycle cycle-indexes 
-                         { pox-addr: pox-addr, first-reward-cycle: first-reward-cycle, num-cycles: num-cycles, amount-ustx: amount-ustx, i: u0 })))
-     (err ERR_STACKING_UNREACHABLE))
-    (ok true)))
+    (asserts! (is-eq num-cycles (get i results)) (err ERR_STACKING_UNREACHABLE))
```

Comment:
Should we replace `(err ERR_STACKING_UNREACHABLE)` with `unwrap-panic`, now that we have the chance?

Reply:
Sure

Reply:
Hmm -- Clarity doesn't have a unconditional `panic` or panicking assert method, so the way to make this panic would be something like:

```
(unwrap-panic (if (is-eq num-cycles (get i results)) (some true) none)
```

or similar. I'm not sure if that actually increases the readability here.

Reply:
You could just `unwrap-panic` the `asserts!`

Reply:
No, the `asserts!` exits the control-flow. It's an early return.

Reply:
Ah, right :facepalm: 

This is probably fine as-is then.